### PR TITLE
Version 2 Testing

### PR DESCRIPTION
**Pull request by khardsonhurley**
- I have started this pull request for implementing tests for version 2 of the library. Currently I have added two to verify user functionality.
- I also added a function called `recurse_dict` in `common.py` which replaces the resource `self` with `self_` as not to conflict with the python `self`.

### CODE REVIEW

**Comment by khardsonhurley** [comment]
@rossdylan Here is the newest pull request, not sure how to add you as a subscriber.

File: tests/client_test.py [hunk 1]
```diff
@@ -2,8 +2,13 @@
...
 import pygerduty
+import pygerduty.v2
 import pytest
 
+###################
+# Version 1 Tests #
+###################
+
 @httpretty.activate
 def test_unknown_subdomain():
```
Comment (khardsonhurley) [inline tests/client_test.py]: Should the code in the body below be 207? Or is 2007 an arbitrary number?

File: pygerduty/common.py [hunk 2]
```diff
@@ -69,9 +69,28 @@ def execute_request(self, request):
         except ValueError:
             response = None
 
+        response = recurse_dict(response)
+
         return response
 
 
+def recurse_dict(d):
```
Comment (gmjosack) [inline pygerduty/common.py]: Can you give this a more descriptive name, like `clean_response`? The current name is more "How it does it" and not "What it does"

**Comment by khardsonhurley** [comment]
@gmjosack @rossdylan The user_test doc is getting quite long with all of the same users. Do you think it would be best to move these into a separate file and import them?

File: tests/schedule_test.py [hunk 3]
```diff
@@ -3,15 +3,36 @@
...
+
+@httpretty.activate
+def test_list_schedules_v2():
+    body = open('tests/fixtures/schedule_list_v2.json').read()
+    httpretty.register_uri(
+	    httpretty.GET, "https://api.pagerduty.com/schedules",
+	    body=body, status=200)
+    p = pygerduty.v2.PagerDuty("contosso", "password")
+    schedules = []
+    for s in p.schedules.list():
```
Comment (khardsonhurley) [inline tests/schedule_test.py]: @gmjosack when I do this, the test goes into an infinite while loop in the list() method. Here is sample output when I put print statements in:
```
1
limit is: 25
offset is: 2486
[<Schedule: schedule_users=[<ScheduleUser: self_=u'https://api.pagerduty.com/users/PEYSGVF', id=u'PEYSGVF', type=u'user', html_url=u'https://subdomain.pagerduty.com/users/PEYSGVF', summary=u'PagerDuty Admin'>], name=u'Daily Engineering Rotation', html_url=u'https://subdomain.pagerduty.com/schedules/PI7DH85', summary=u'Daily Engineering Rotation', self_=u'https://api.pagerduty.com/schedules/PI7DH85', time_zone=u'America/New_York', escalation_policies=[<EscalationPolicy: self_=u'https://api.pagerduty.com/escalation_policies/PT20YPA', id=u'PT20YPA', type=u'escalation_policy_reference', html_url=u'https://subdomain.pagerduty.com/escalation_policies/PT20YPA', summary=u'Another Escalation Policy'>], type=u'schedule', id=u'PI7DH85', description=u'Rotation schedule for engineering'>]
1
limit is: 25
offset is: 2487
[<Schedule: schedule_users=[<ScheduleUser: self_=u'https://api.pagerduty.com/users/PEYSGVF', id=u'PEYSGVF', type=u'user', html_url=u'https://subdomain.pagerduty.com/users/PEYSGVF', summary=u'PagerDuty Admin'>], name=u'Daily Engineering Rotation', html_url=u'https://subdomain.pagerduty.com/schedules/PI7DH85', summary=u'Daily Engineering Rotation', self_=u'https://api.pagerduty.com/schedules/PI7DH85', time_zone=u'America/New_York', escalation_policies=[<EscalationPolicy: self_=u'https://api.pagerduty.com/escalation_policies/PT20YPA', id=u'PT20YPA', type=u'escalation_policy_reference', html_url=u'https://subdomain.pagerduty.com/escalation_policies/PT20YPA', summary=u'Another Escalation Policy'>], type=u'schedule', id=u'PI7DH85', description=u'Rotation schedule for engineering'>]
```

where 1 represents the length of the page, the limit is set to the default and the offset keeps getting incremented infinitely. I did notice that the response output changed for this between v1 and v2 but can't seem to figure out why its getting stuck in the loop and never did that before.

File: pygerduty/v2.py [hunk 4]
```diff
@@ -132,24 +136,26 @@ def list(self, **kwargs):
...
                 for item in this_paginated_result:
                     if item.id in seen_items:
                         continue
                     seen_items.add(item.id)
                     yield item
-                offset += len(this_paginated_result)
-                print len(this_paginated_result)
-                print "limit is: %s" % limit
-                print "offset is: %s" % offset
+                offset -= len(this_paginated_result)
```
Comment (khardsonhurley) [inline pygerduty/v2.py]: @gmjosack @rossdylan Can you both check the logic here? This change allowed `test_list_incidents_v2` to pass, but I am not sure if this is what was originally intended.

**Comment by gmjosack** [comment]
Sorry for the delay here. @khardsonhurley and I worked offline to figure this out, was a mocking bug, over a pygerduty bug.

**Pull request merged**